DYDX-Bots/Large_Borrower_Allocation/src/agent.ts

```diff
@@ -0,0 +1,33 @@
+import { Finding, HandleTransaction, TransactionEvent, getEthersProvider } from "forta-agent";
+import { BigNumber, providers } from "ethers";
+import NetworkData from "./network";
+import NetworkManager from "./network";
+import { THRESHOLD, SCHED_BORROW_ALLOC_CHANGE_EVENT } from "./utils";
+import { createFinding } from "./findings";
+
+const networkManager = new NetworkManager();
+
+export const provideInitialize = (provider: providers.Provider) => async () => {
+  const { chainId } = await provider.getNetwork();
+  networkManager.setNetwork(chainId);
+};
+
+export function provideHandleTransaction(networkManager: NetworkData, threshold: BigNumber): HandleTransaction {
```

Comment:
Any specific reason why the networkManager parameter has a `NetworkData` type not a `NetworkManager`?

Reply:
When I was going off the ApeSwap bot that first implemented it, the type was set as `NetworkData`, and me trying to make sure I got it to work properly, use the same. However, I just switched it to `NetworkManager` to see if it would break, and it still functions as expected. Perhaps they can be interchangeable? Might make sense since the `NetworkManager` class `implements` the `NetworkData` interface.

Reply:
`NetworkData` is correct, in fact the idea is to use a `NetworkData` you can even change the name of the variable, because the int terms of the bot logic it only needs a NetworkData, because it is not going to Manage the network anyway.